### PR TITLE
Merge branch-DeveloperGuide-Qiaoran with Master

### PR DESCRIPTION
**Pull request by Qiaoran-M**

### CODE REVIEW

File: docs/DeveloperGuide.md
Comment (Benjamin-Sim) [inline docs/DeveloperGuide.md]: Perhaps format as code?

```suggestion
person command text entered by user. The specific `UniquePersonList` operations are not shown in the diagram
```

File: docs/DeveloperGuide.md
Comment (Benjamin-Sim) [inline docs/DeveloperGuide.md:209]: I think you should include the return arrow at the end of the activation bar for `:LogicManager`

File: docs/DeveloperGuide.md
Comment (Benjamin-Sim) [inline docs/DeveloperGuide.md]: Small grammar change

```suggestion
activities related to gender field are considered and shown in this activity diagram.
```

File: docs/DeveloperGuide.md
Comment (Benjamin-Sim) [inline docs/DeveloperGuide.md]: Small grammar change

```suggestion
<div markdown="span" class="alert alert-info">:information_source: **Note:**Parser exceptions are thrown and caught if 
```

File: docs/DeveloperGuide.md
Comment (Benjamin-Sim) [inline docs/DeveloperGuide.md]: Small grammar change

```suggestion
gender field is not provided in the command, or the gender is not of valid format; Duplicated person exception is 
```

File: docs/DeveloperGuide.md
Comment (Benjamin-Sim) [inline docs/DeveloperGuide.md:218]: I believe the notation used for branching paths is incorrect, see [here](https://nus-cs2103-ay2223s1.github.io/website/schedule/week9/topics.html#design-modelling-modelling-behaviors-activity-diagrams-basic). The branch conditions should be in square brackets next to the arrows leading away from the branch node.

File: docs/DeveloperGuide.md
Comment (Benjamin-Sim) [inline docs/DeveloperGuide.md:254]: Same as the add person sequence diagram, I think you should include the return arrow at the end of the activation bar for `:LogicManager`

File: docs/DeveloperGuide.md
Comment (Benjamin-Sim) [inline docs/DeveloperGuide.md]: Perhaps format as code?

```suggestion
person command text entered by user; the `setPerson(P1, P2)` method replaces person P1 with person P2 in the person list 
```

File: docs/DeveloperGuide.md
Comment (Benjamin-Sim) [inline docs/DeveloperGuide.md]: Perhaps format as code?

```suggestion
in the model. The specific `UniquePersonList` operations are not shown in the diagram for simplicity.
```

File: docs/DeveloperGuide.md
Comment (Benjamin-Sim) [inline docs/DeveloperGuide.md:263]: Same issue as the add person activity diagram

File: docs/DeveloperGuide.md
Comment (Benjamin-Sim) [inline docs/DeveloperGuide.md]: Small grammar change
```suggestion
activities related to gender field are considered and shown in this activity diagram. All fields are considered optional 
```

File: docs/DeveloperGuide.md
Comment (Benjamin-Sim) [inline docs/DeveloperGuide.md]: Small grammar change
```suggestion
<div markdown="span" class="alert alert-info">:information_source: **Note:**Parser exceptions are thrown and caught if 
```

File: docs/DeveloperGuide.md
Comment (Benjamin-Sim) [inline docs/DeveloperGuide.md]: Should change this back to the original text

```suggestion
### \[Proposed\] Undo/redo feature
#### Proposed Implementation
```

**Review by Benjamin-Sim — CHANGES_REQUESTED** [review]
Cool sequence diagrams! They look better than the ones in the original developer guide 😄
The activity diagrams look good too but needs some rework due to incorrect notation.

Besides that it's mostly good, just some minor edits.

Also they recommend you version control the `.puml` files for the diagrams as stated [here](https://nus-cs2103-ay2223s1.github.io/website/schedule/week10/project.html#admin-tp-deliverables-dg-tips).

Lastly, I notice that you refer to the classes `AddCommand` and `EditCommand` in the text and sequence diagrams. Do you want to write `AddPersonCommand` and `EditPersonCommand` instead since we're going to rename the classes?

File: docs/DeveloperGuide.md
Comment (Qiaoran-M) [reply docs/DeveloperGuide.md:209]: Thanks for all the suggestions! I will change the diagrams and add in the source files

File: docs/DeveloperGuide.md
Comment (tanhl2000) [inline docs/DeveloperGuide.md:263]: The text for the conditions `gender format is valid` and `Person already exists in the contact list` is not properly contained inside the conditional cell. Same issues for add person activity diagram.

File: docs/DeveloperGuide.md
Comment (tanhl2000) [inline docs/DeveloperGuide.md:209]: For the model section, you put the type as `:model` however `model` is the interface that is implemented by `modelManager`, as such I think it is better to put the type as `modelManager` instead.

Comment (tanhl2000) [inline docs/DeveloperGuide.md:209]: Also do you want to add the `AddPersonCommandParser` in between the `AddressBookParser` and `AddPersonCommand`? I feel that this interaction is important to be shown.

**Review by tanhl2000 — CHANGES_REQUESTED** [review]
Other than these small issues looks good!

File: docs/DeveloperGuide.md
Comment (Benjamin-Sim) [inline docs/DeveloperGuide.md]: Since the sequence diagram uses `AddPersonCommand`, this line should be updated to match as well.

```suggestion
 of `addPersonCommand#execute()`.
```

File: docs/DeveloperGuide.md
Comment (Benjamin-Sim) [inline docs/DeveloperGuide.md]: Missing brackets for method

```suggestion
`AddressBook#addPerson()` which calls `UniquePersonList#add()` to add a new person in the person list
```

**Review by tanhl2000 — APPROVED** [review]
Looks good for changes i requested, wait for @Benjamin-Sim to approve as well before merging!

File: docs/DeveloperGuide.md
Comment (Benjamin-Sim) [reply docs/DeveloperGuide.md:209]: `AddPersonCommandParser()` method creates a new `AddPersonCommand` object. So the sequence diagram should show an arrow from the `AddPersonCommandParser` activation bar to the `AddPersonCommand` box instead of its activation bar.

Also there is an arrow from the `AddPersonCommand` activation bar pointing to the box. I think it should be a dotted line since that's the lifeline

Capture

File: docs/DeveloperGuide.md
Comment (Benjamin-Sim) [reply docs/DeveloperGuide.md:218]: Hmm... the notation for the branch nodes is still incorrect. The diamond branch node should be empty, while the branch conditions (e.g. "gender field is provided") should be placed beside the arrows in square brackets. Therefore there is no need for the "yes" and "no" labels.

Refer to the [textbook](https://nus-cs2103-ay2223s1.github.io/website/schedule/week9/topics.html#design-modelling-modelling-behaviors-activity-diagrams-basic) to see how it should be drawn

File: docs/DeveloperGuide.md
Comment (Benjamin-Sim) [inline docs/DeveloperGuide.md]: Same as the `AddPersonCommand`, this line should be updated to match the diagram.

```suggestion
It is performed as a part of `editPersonCommand#execute()`.
```

File: docs/DeveloperGuide.md
Comment (Benjamin-Sim) [inline docs/DeveloperGuide.md]: Missing brackets for method

```suggestion
`AddressBook#setPerson()` which calls `UniquePersonList#setPerson()` to replace an existing person with a new person
```

File: docs/DeveloperGuide.md
Comment (Benjamin-Sim) [reply docs/DeveloperGuide.md:254]: Similar to the add person sequence diagram, the `EditPersonCommandParser()` method creates a new `EditPersonCommand` object. So the sequence diagram should show an arrow from the `EditPersonCommandParser` activation bar to the `EditPersonCommand` box instead of its activation bar.

File: docs/DeveloperGuide.md
Comment (Benjamin-Sim) [reply docs/DeveloperGuide.md:263]: Same issue as the add person activity diagram, the branch node should be changed

**Review by Benjamin-Sim — CHANGES_REQUESTED** [review]
Just a few more edits 👍

**Review by Benjamin-Sim — CHANGES_REQUESTED** [review]
Activity diagrams are good 👍, but the sequence diagrams have 1 more change.

The arrow going to the `AddPersonCommand` box should come from the activation bar of `AddPersonCommandParser`:

Capture

Same thing for `EditPersonCommand`.

Sorry I know I'm being picky about these diagrams but I think it's crucial that we get these basic notation issues correct. Otherwise they become easy-to-find bugs that the graders can mark us down on. 😄

**Review by Benjamin-Sim — APPROVED** [review]
LGTM! 👍

Note: For the add person sequence diagram, the `parse(cmd)` method is actually a self-invocation of the `AddPersonCommandParser` object. The code actually creates the `AddPersonCommandParser` object first, then calls its `parse()` method. So it is only within that self-invocation that the constructor of `AddPersonCommand` is called.

Same thing with the edit person sequence diagram. I only just noticed this after looking at the code. 😭

I'll leave it to you if you want to change it becuae I think this is a minor enough issue to let it slide. 😅

**Pull request merged**
